include/lsst/afw/geom/SphPoint.h

```diff
@@ -0,0 +1,319 @@
+     * This class was not designed for inheritance, and must not be subclassed.
+     *
+     * @see @ref coord::Coord
+     */
+    // TODO: add final keyword after migrating to Pybind11
+    class SphPoint
+    {
+    public:
+        /**
+         * @brief Represent a specific pair of spherical coordinates.
+         *
+         * @param longitude The azimuth angle of the point.
+         * @param latitude The elevation angle of the point. Must correspond
+         *                 to an angle in [-&pi;/2, &pi;/2] radians.
+         *
+         * @throws InvalidParameterError Thrown if any argument is NaN or
+         *                               infinite.
+         * @throws OutOfRangeError Thrown if @c latitude is out of range.
+         *
+         * @par Exception Safety:
+         * The program state shall be unchanged in the event of an exception.
+         */
+        SphPoint(Angle const & longitude, Angle const & latitude);
+
+        /**
+         * @brief Represent a specific direction in 3-dimensional space.
+         *
+         * The adopted convention is that <1, 0, 0> corresponds to (0, 0), that
+         * <0, 1, 0> corresponds to (&pi;/2, 0), and that <0, 0, 1> corresponds
+         * to (0, &pi;/2).
+         *
+         * @param vector A position whose direction will be stored as a point.
+         *               Must not be the zero vector. Need not be normalized.
+         *
+         * @throws InvalidParameterError Thrown if any element of @c vector is
+         *                               NaN or infinite, or if @c vector is
+         *                               the zero vector.
+         *
+         * @par Exception Safety:
+         * The program state shall be unchanged in the event of an exception.
+         */
+        explicit SphPoint(Point3D const & vector);
+
+        /**
+         * @brief Creates a copy of a SphPoint.
+         *
+         * @param other The point to copy.
+         *
+         * @par Exception Safety:
+         * The program state shall be unchanged in the event of an exception.
+         */
+        SphPoint(SphPoint const & other) = default;
+
+        /**
+         * Overwrites this object with the value of another SphPoint.
+         *
+         * This is the only method that can alter the state of a SphPoint after
+         * its construction, and is provided to allow in-place replacement of
+         * SphPoints where swapping is not possible.
+         *
+         * @param other The object with which to overwrite this one.
+         * @return a reference to this object.
+         *
+         * @par Exception Safety:
+         * This operator shall not throw exceptions.
+         */
+        SphPoint& operator=(SphPoint const & other) = default;
+
+        /*
+         * Accessors
+         */
+
+        /**
+         * @brief The azimuth angle of this point.
+         *
+         * If this point is at a coordinate pole, this method may return any
+         * value.
```

Comment:
I saw the original comment as merely a matter of being as explicit as possible in the spec -- there is no such thing as "the" longitude of a polar point, so the implementation is free to choose any value that conforms to the [0, 2π) requirement. It could be whatever was passed to the constructor (the current implementation), it could always be zero, it could be some magic value that makes internal calculations more efficient, etc.

My question is, why should the user *care* about the longitude of a polar SphPoint? It won't affect the result of `separationTo` or `rotated` (or, at least, it shouldn't -- that would be a good test case), it won't affect equality comparisons, etc.. It doesn't even affect the singularity associated with `bearingTo`, because that's caused by the absence of well-defined directions at the pole, not the absence of well-defined longitude.

The only thing it does affect is the value of `getLongitude()` (and synonyms like `[0]`), and if you're using those methods because you want to do coordinate math yourself, then it's up to you to make sure your equations are valid in spherical geometry.